crates/edr_rpc_eth/src/receipt.rs

```diff
@@ -186,6 +222,64 @@ impl TryFrom<Block> for receipt::BlockReceipt<TypedEnvelope<receipt::Execution<F
     }
 }
 
+impl TryFrom<Block>
+    for receipt::BlockReceipt<TypedEnvelope<receipt::execution::Eip658<FilterLog>>>
+{
+    type Error = ConversionError;
+
+    fn try_from(value: Block) -> Result<Self, Self::Error> {
+        let transaction_type = value
+            .transaction_type
+            .map_or(Ok(transaction::Type::Legacy), transaction::Type::try_from)
+            .map_err(ConversionError::UnknownType)?;
+
+        let execution = if transaction_type == transaction::Type::Legacy {
+            if let Some(status) = value.status {
+                receipt::execution::Eip658 {
+                    status,
+                    cumulative_gas_used: value.cumulative_gas_used,
+                    logs_bloom: value.logs_bloom,
+                    logs: value.logs,
+                }
+            } else if let Some(root) = value.state_root {
+                receipt::execution::Legacy {
+                    root,
+                    cumulative_gas_used: value.cumulative_gas_used,
+                    logs_bloom: value.logs_bloom,
+                    logs: value.logs,
+                }
+                .into()
```

Comment:
@fvictorio this is a product change that converts a remote legacy receipt to an internal EIP-658 receipt. As a result, we never report the `root` anymore and always report a `status` of `1`/`true`.